entities/storobj/storage_object.go

```diff
@@ -432,26 +434,80 @@ func (ko *Object) MarshalBinary() ([]byte, error) {
 // UnmarshalPropertiesFromObject only unmarshals and returns the properties part of the object
 //
 // Check MarshalBinary for the order of elements in the input array
-func UnmarshalPropertiesFromObject(data []byte, properties *models.PropertySchema) error {
+func UnmarshalPropertiesFromObject(data []byte, properties *map[string]interface{}, aggregationProperties []string, propStrings [][]string) error {
 	if data[0] != uint8(1) {
 		return errors.Errorf("unsupported binary marshaller version %d", data[0])
 	}
 
+	// clear out old values in case an object misses values. This should NOT shrink the capacity of the map, eg there
+	// are no allocations when adding the properties of the next object again
+	for k := range *properties {
+		delete(*properties, k)
+	}
+
 	startPos := uint64(1 + 8 + 1 + 16 + 8 + 8) // elements at the start
 	byteOps := byte_operations.ByteOperations{Position: startPos, Buffer: data}
 	// get the length of the vector, each element is a float32 (4 bytes)
 	vectorLength := uint64(byteOps.ReadUint16())
 	byteOps.MoveBufferPositionForward(vectorLength * 4)
 
-	// length of class name
 	classnameLength := uint64(byteOps.ReadUint16())
 	byteOps.MoveBufferPositionForward(classnameLength)
-
-	// property schema length
 	propertyLength := uint64(byteOps.ReadUint32())
-	if err := json.Unmarshal(data[byteOps.Position:byteOps.Position+propertyLength], properties); err != nil {
-		return err
-	}
+
+	jsonparser.EachKey(data[byteOps.Position:byteOps.Position+propertyLength], func(idx int, value []byte, dataType jsonparser.ValueType, err error) {
+		switch dataType {
+		case jsonparser.Number:
+			v, _ := jsonparser.ParseFloat(value)
+			(*properties)[aggregationProperties[idx]] = v
+		case jsonparser.Boolean:
+			b, _ := jsonparser.ParseBoolean(value)
+			(*properties)[aggregationProperties[idx]] = b
+		case jsonparser.String:
+			s, _ := jsonparser.ParseString(value)
+			(*properties)[aggregationProperties[idx]] = s
+		case jsonparser.Array: // can be a beacon or an actual array
+			ArrayEntries := value[1 : len(value)-1] // without leading and trailing []
```

Comment:
I think this variable should be renamed to `arrayEntries` to be private one

Reply:
renamed, but it's a local variable so shouldn't make a difference?

Reply:
yeah, it shouldn't, but it's more of a convention that the private variables start with a small letter